internal/controller/cloudstorage_controller.go

```diff
@@ -128,41 +139,91 @@ func (b CloudStorageReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		created, err := clnt.Create()
-		if !created {
-			logger.Info("unable to create object bucket")
+		if !created || err != nil {
+			logger.Info("unable to create object bucket", "error", err)
 			b.EventRecorder.Event(&bucket, corev1.EventTypeWarning, "BucketNotCreated", fmt.Sprintf("unable to create bucket: %v", err))
-			return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
-		}
-		if err != nil {
-			//TODO: LOG/EVENT THE MESSAGE
-			logger.Error(err, "Error while creating event")
-			return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+			// Set condition and return error to trigger exponential backoff
+			apimeta.SetStatusCondition(&bucket.Status.Conditions, metav1.Condition{
+				Type:    oadpv1alpha1.ConditionBucketReady,
+				Status:  metav1.ConditionFalse,
+				Reason:  oadpv1alpha1.ReasonBucketCreationFailed,
+				Message: fmt.Sprintf("Failed to create bucket: %v", err),
+			})
+			bucket.Status.LastSynced = &metav1.Time{Time: time.Now()}
+			bucket.Status.Name = bucket.Spec.Name
+			if updateErr := b.Client.Status().Update(ctx, &bucket); updateErr != nil {
+				logger.Error(updateErr, "failed to update CloudStorage status")
+			}
+			// Return error to trigger exponential backoff
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, fmt.Errorf("bucket creation failed")
 		}
+		// Bucket created successfully
 		b.EventRecorder.Event(&bucket, corev1.EventTypeNormal, "BucketCreated", fmt.Sprintf("bucket %v has been created", bucket.Spec.Name))
-	}
-	if err != nil {
-		// Bucket may be created but something else went wrong.
-		logger.Error(err, "unable to determine if bucket exists.")
-		b.EventRecorder.Event(&bucket, corev1.EventTypeWarning, "BucketNotFound", fmt.Sprintf("unable to find bucket: %v", err))
-		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+		apimeta.SetStatusCondition(&bucket.Status.Conditions, metav1.Condition{
+			Type:    oadpv1alpha1.ConditionBucketReady,
+			Status:  metav1.ConditionTrue,
+			Reason:  oadpv1alpha1.ReasonBucketCreated,
+			Message: fmt.Sprintf("Bucket %v has been created successfully", bucket.Spec.Name),
+		})
+	} else if err != nil {
+		// Error checking if bucket exists
+		logger.Error(err, "unable to determine if bucket exists")
+		b.EventRecorder.Event(&bucket, corev1.EventTypeWarning, "BucketNotFound", fmt.Sprintf("unable to check bucket: %v", err))
+		apimeta.SetStatusCondition(&bucket.Status.Conditions, metav1.Condition{
+			Type:    oadpv1alpha1.ConditionBucketReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  oadpv1alpha1.ReasonBucketCheckError,
+			Message: fmt.Sprintf("Unable to verify bucket status: %v", err),
+		})
+		bucket.Status.LastSynced = &metav1.Time{Time: time.Now()}
+		bucket.Status.Name = bucket.Spec.Name
+		if updateErr := b.Client.Status().Update(ctx, &bucket); updateErr != nil {
+			logger.Error(updateErr, "failed to update CloudStorage status")
+		}
+		// Return error to trigger exponential backoff
+		return ctrl.Result{}, err
```

Comment:
`return ctrl.Result{}, err` makes it requeue with exponential backoff if `err != nil`.